File: typescript/base-hmi-types/src/screens/controls/HmiAlarmColumn.ts
import { HmiMultilingualText } from "../../common/HmiMultilingualText.js";
import { HmiProperty } from "../base.js";
import { HmiAlarmColumnType } from "./HmiAlarmColumnType.js";

export class HmiAlarmColumn {
  type = HmiAlarmColumnType.Unknown;
  sourceType?: string;
  visible?: HmiProperty<boolean>;
  timeAndDateFormat?: string;
  headerText?: HmiMultilingualText;
  symbol?: string;
}

File: src/BaseHmiTypes/Screens/Controls/HmiAlarmColumn.cs
using BaseHmiTypes.Common;
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Controls;

public sealed class HmiAlarmColumn
{
    public HmiAlarmColumnType Type { get; set; }
    public string? SourceType { get; set; }
    public HmiProperty<bool>? Visible { get; set; }

    public HmiProperty<double>? Width { get; set; }
    public string? TimeAndDateFormat { get; set; }
    public HmiMultilingualText? HeaderText { get; set; }
    public string? Symbol { get; set; }
}
